postgres/datadog_checks/postgres/postgres.py

```diff
@@ -142,7 +144,9 @@ def _run_query_scope(self, cursor, scope, is_custom_metrics, relations_config, c
             # This happens for example when trying to get replication metrics from readers in Aurora. Let's ignore it.
             log_func(e)
             self.db.rollback()
-            self._is_aurora = None
+            self.log.debug("Disabling replication metrics")
+            self._is_aurora = False
+            self.metrics_cache.replication_metrics = {}
```

Comment:
Setting it to None without also setting replication_metrics does nothing because replication_metrics were not being calculated again.